File: class4/16953 a to b.py
# A -> B
# 16953 

a, b = map(int, input().split())

# 최대 50
# 가지를 어떻게 표현하지 

rst = -1

def recursion(result, goal, depth):
    if result > goal:
        return
    elif result == goal:
        global rst
        rst = depth+1
        return

    recursion(result*2, goal, depth+1)
    recursion(result*10+1, goal, depth+1)


recursion(a, b, 0)
print(rst)

File: class4/16953 a to b.c
#include <stdio.h>
#pragma waring (disable: 4996)


int recursion(int result, int goal, int depth);

int main(){
    int a, b;

    scanf("%d %d", &a, &b);
    printf("%d", recursion(a, b, 0) + 1);

    return 0;

}

int recursion(int result, int goal, int depth){
    if (result>goal) return -1;
    else if (result == goal) return depth;

    recursion(result*2, goal, depth+1);
    recursion(result*10+1, goal, depth+1);

}
